src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -862,6 +682,59 @@ def hcells_list(self, value: List[str]) -> None:
     ### Outputs ###
     ### END Generated interface HammerLVSTool ###
 
+class HasUPFSupport(HammerTool):
+    """Mix-in trait with functions useful for tools with UPF style power
+    constraints"""
+    @property
+    def upf_power_specification(self) -> str:
+        raise NotImplementedError("Automatic generation of UPF power specifications is not supported yet.")
+
+class HasCPFSupport(HammerTool):
```

Comment:
No changes requested here, but adding this to #184 as a list of things we want to eventually move. CPF is Cadence-specific and IMO should live with the rest of the Cadence stuff.

Reply:
There is a spec that is available online. So I think its a bit different than the ILM stuff. https://projects.si2.org/?page=811